src/mlpack/methods/reinforcement_learning/environment/bit_flipping.hpp

```diff
@@ -0,0 +1,256 @@
+/**
+ * @file methods/reinforcement_learning/environment/bit_flipping.hpp
+ * @author Eshaan Agarwal
+ *
+ * This file is an implementation of Bit Flipping toy task:
+ * https://github.com/NervanaSystems/gym-bit-flip/blob/master/gym_bit_flip/bit_flip.py
+ * https://github.com/ceteke/her/blob/master/env.py
+ *
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_RL_ENVIRONMENT_BIT_FLIPPING_HPP
+#define MLPACK_METHODS_RL_ENVIRONMENT_BIT_FLIPPING_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+
+/**
+ * Implementation of Cart Pole task.
+ */
+class BitFlipping
+{
+ public:
+  /**
+   * Implementation of the state of Bit Flipping. 
+   * Each state is a n length binary vector
+   */
+  class State
+  {
+   public:
+    /**
+     * Construct a state instance.
+     */
+    State() : data(dimension)
+    { /* Nothing to do here. */ }
+
+    /**
+     * Construct a state instance from given data.
+     *
+     * @param data Data 
+     */
+    State(const arma::vec& data) : data(data)
+    { /* Nothing to do here */ }
+
+    //! Modify the internal representation of the state.
+    arma::vec& Data() { return data; }
+
+    //! Get the internal representation of the state.
+    arma::vec Data() const { return data; }
+
+    //! Encode the state to a column vector.
+    const arma::vec& Encode() const { return data; }
+
+    //! Dimension of the encoded state.
+    static constexpr size_t dimension = 2;
+
+   private:
+    //! Locally-stored n bit integer.
+    arma::vec data;
+  };
+
+  /**
+   * Implementation of action of Bit Flipping.
+   */
+  class Action
+  {
+   public:
+    
+    // To store the action ( index of n bit number to be flipped )
+    size_t action;
+
+    // Track the size of the action space.
+    static const size_t size = 1;
+  };
+
+  /**
+   * Construct a binary vector instance using the given constants.
+   *
+   * @param maxSteps The number of steps after which the episode
+   *    terminates. If the value is 0, there is no limit.
+   * @param length Length of the binary vector for state and goal
+   */
+  BitFlipping(const size_t maxSteps = 200,
+              const size_t length = 2) :
+      maxSteps(maxSteps),
+      length(length),
+      stepsPerformed(0)
+  { 
+    
+  }
+
+  /**
+   * Get reward and next state based on current
+   * state and current action.
+   *
+   * @param state The current state.
+   * @param action The current action.
+   * @param nextState The next state.
+   * @param transitionGoal The transition goal.
+   * @return reward, it's always 1.0.
```

Comment:
Why is this always 1? I can see you return 0 at the end.

Reply:
I will make that change. I took it from other environmenets.